src/IssueViz.Security/Taint/TaintIssuesSynchronizer.cs

```diff
@@ -166,10 +166,14 @@ private bool IsFeatureSupported(ServerInfo serverInfo)
             return false;
         }
 
-        private async Task<AnalysisInformation> GetAnalysisInformation(string projectKey)
+        private async Task<AnalysisInformation> GetAnalysisInformation(string projectKey, string branchName)
         {
             var branches = await sonarQubeService.GetProjectBranchesAsync(projectKey, CancellationToken.None);
-            var issuesBranch = branches.First(x => x.IsMain);
+
+            var issuesBranch = string.IsNullOrEmpty(branchName)
```

Comment:
With the recent change I did, branchName should not be null anymore, but still, I figured I might as well leave this. WDYT?

Reply:
If we'd documented that our API should never return `null`, then we could `Debug.Assert` instead - a `null` would mean our code isn't working as expected.

Reply:
So what is the behavior we expect if we can't find a matching branch? I guess we could throw and clear the store 🤔

Reply:
It should never happen; if it does, it's because there's a bug in another part of our code and we're returning `null` when we said we wouldn't.

If it were third-party code then I'd be defensive and handle it being `null`, since we're not in control of the behaviour.

Here, we are in control, so I wouldn't be defensive. The assertion documents what we expect from our code. If we've messed it up the user will get a runtime failure.